woocommerce/woocommerce-ios
# [Variations] Display empty state when navigating to variation list

#closes #3195

# Why

This PR adds the entry point for allowing a merchant to add a product's first variation.
This is done by displaying an empty state screen when tapping on the "Add Variation" cell that has an "Add variation" button that will allow the merchant to initiate the add variation flow, whose first step is to add an attribute(screen in progress)

# How

- Added `emptyBoxImage` resource
- Update `EmptyStateViewController` to add a way to provide a custom native action when tapping the action button
- Update `ProductFormViewController` to always navigate to variations list when tapping the variations cell.
- Update `ProductVariationsViewController ` to show the `EmotyStateScreen` when there are no variations.
- Update `ProductVariationsViewController ` to navigate to `AddAttributeViewController` when tapping on `Add variation` button

# Gifs

No Variations | Existing Variations
---- | ----
![no-variations](https://user-images.githubusercontent.com/562080/106407425-fc946900-6409-11eb-81b2-cdb59ee0b0b9.gif) | ![existing-variations](https://user-images.githubusercontent.com/562080/106407429-0027f000-640a-11eb-96ea-e0e7abf7832b.gif)


# Testing Steps
- Navigate to a variable product that has existing variations
- Tap on the variations cell
- See that the variations list is displayed

- Navigate to a variable product without variations
- Tap on the "Add variation" cell
- See that the empty variations screen is displayed
- Tap on the "Add variation" button
- See that it navigates to the "Add Attribute" screen.

Update release notes:
